modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/portlet/action/ExpireObjectEntryMVCActionCommand.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.web.internal.object.entries.portlet.action;
+
+import com.liferay.object.model.ObjectEntry;
+import com.liferay.object.rest.resource.v1_0.ObjectEntryResource;
+import com.liferay.object.service.ObjectEntryLocalService;
+import com.liferay.portal.kernel.feature.flag.FeatureFlagManagerUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.Objects;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+/**
+ * @author Jhosseph Gonzalez
+ */
+public class ExpireObjectEntryMVCActionCommand extends BaseMVCActionCommand {
+
+	public ExpireObjectEntryMVCActionCommand(
+		ObjectEntryLocalService objectEntryLocalService,
+		ObjectEntryResource.Factory objectEntryResourceFactory) {
+
+		_objectEntryLocalService = objectEntryLocalService;
+		_objectEntryResourceFactory = objectEntryResourceFactory;
+	}
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-17564")) {
+			throw new UnsupportedOperationException();
+		}
+
+		ObjectEntryResource.Builder builder =
+			_objectEntryResourceFactory.create();
+
+		long objectEntryId = ParamUtil.getLong(actionRequest, "objectEntryId");
+
+		ObjectEntry serviceBuilderObjectEntry =
+			_objectEntryLocalService.getObjectEntry(objectEntryId);
+
+		if (!Objects.equals(
+				serviceBuilderObjectEntry.getStatus(),
+				WorkflowConstants.STATUS_DRAFT) &&
+			!Objects.equals(
+				serviceBuilderObjectEntry.getStatus(),
+				WorkflowConstants.STATUS_PENDING)) {
+
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)actionRequest.getAttribute(WebKeys.THEME_DISPLAY);
+
+			ObjectEntryResource objectEntryResource = builder.user(
```

Comment:
Is there a specific reason to use resource layer here? If not, use ObjectEntryService

Reply:
@gabrielwas  While explaining how to do this,  Marcela said it would be better to create a new method in the resource layer and use it there, since it’s adjacent to the MVC layer. If that’s not the case, please let me know so I can update it now.

Reply:
@marcelabc could you give your thoughts here, please?